src/wp-includes/class-wp-term-query.php

```diff
@@ -728,7 +719,12 @@ public function get_terms() {
 		}
 
 		// $args can be anything. Only use the args defined in defaults to compute the key.
-		$key          = md5( serialize( wp_array_slice_assoc( $args, array_keys( $this->query_var_defaults ) ) ) . serialize( $taxonomies ) . $this->request );
+		$cache_args = wp_array_slice_assoc( $args, array_keys( $this->query_var_defaults ) );
+		unset( $cache_args['pad_counts'], $cache_args['update_term_meta_cache'] );
+		if ( 'count' !== $_fields && 'all_with_object_id' !== $_fields ) {
+			$cache_args['fields'] = 'all';
+		}
+		$key          = md5( serialize( $cache_args ) . serialize( $taxonomies ) . $this->request );
```

Comment:
```suggestion
		ksort( $cache_args );
		$key          = md5( serialize( $cache_args ) . serialize( $taxonomies ) . $this->request );
```

This is a maybe, what do you think?

Reply:
The existing tests prove that it is not needed, no?

Reply:
I couldn't see an existing test that does, turns out it's not needed because of the way `array_merge` works.

It would be good to test in case something changes, either within the class or the way PHP works.

I'll add them as a suggestion.